eq-author/src/components/Comments/Comment/index.js

```diff
@@ -0,0 +1,320 @@
+              commentId: id,
+              commentText,
+            },
+          },
+        });
+      }
+    },
+    [id, subjectId, rootId, isReply]
+  );
+
+  const onDeleteComment = useCallback(() => {
+    if (isReply) {
+      deleteReply({
+        variables: {
+          input: {
+            componentId: subjectId,
+            commentId: rootId,
+            replyId: id,
+          },
+        },
+      });
+    } else {
+      deleteComment({
+        variables: {
+          input: {
+            componentId: subjectId,
+            commentId: id,
+          },
+        },
+      });
+    }
+  }, [id, subjectId, rootId, isReply]);
+
+  return (
+    <Wrapper data-test="Comment">
+      <Header data-test="Comment__Header">
+        <Avatar data-test="Comment__Avatar">{authorInitials}</Avatar>
+        <ColumnWrapper>
+          <Author data-test="Comment__Author">{authorName}</Author>
+          <Date data-test="Comment__DatePosted">
+            {moment(datePosted).calendar()}
+          </Date>
+        </ColumnWrapper>
+        <RightButtonGroup>
+          {canEdit && (
+            <IconButton
+              data-test="Comment__EditCommentBtn"
+              icon={iconEdit}
+              onClick={() => {
+                setEditing(true);
+              }}
+            >
+              Edit comment
+            </IconButton>
+          )}
+
+          {canDelete && (
+            <IconButton
+              data-test="Comment__DeleteCommentBtn"
+              icon={iconClose}
+              onClick={() => onDeleteComment()}
+            >
+              Delete comment
+            </IconButton>
+          )}
+        </RightButtonGroup>
+      </Header>
+      <Body data-test="Comment__Body">
+        {editing ? (
+          <CommentEditor
+            data-test="Comment__CommentEditor"
+            canClose
+            confirmText={"Save"}
+            initialValue={commentText}
+            variant={"growable"}
+            onConfirm={(commentText) => {
+              onUpdateComment(commentText);
+              setEditing(false);
+            }}
+            onCancel={() => {
+              setEditing(false);
+            }}
+          />
+        ) : (
+          <Text data-test="Comment__CommentText">{commentText}</Text>
+        )}
+        {dateModified && (
+          <Date data-test="Comment__DateModified">{`Edited: ${moment(
+            dateModified
+          ).calendar()}`}</Date>
+        )}
+      </Body>
+    </Wrapper>
+  );
+};
+
+IconButton.propTypes = {
+  icon: PropTypes.string.isRequired,
+  onClick: PropTypes.func.isRequired,
+  children: PropTypes.node.isRequired,
+  "data-test": PropTypes.string,
+};
+
+Comment.propTypes = {
+  /**
+   * The ID of the comment.
+   */
+  id: PropTypes.string.isRequired,
+  /**
+   * The ID of the comment at the root of the chain.
+   * If the comment is the one at the root of the chain,
+   * the value of this will be equal to ID.
+   */
```

Comment:
are you happy leaving these descriptive comments in? Most of these propNames are pretty self descriptive